### PR TITLE
Allow storing non boolean value to localStorage

### PR DESCRIPTION
**Pull request by zhouyx**
As discussed in #17742, this is pre work to support storing consent string.

### CODE REVIEW

File: build-system/tasks/presubmit-checks.js [hunk 1]
```diff
@@ -836,6 +836,12 @@ const forbiddenTermsSrcInclusive = {
       'src/service/url-replacements-impl.js',
     ],
   },
+  '\\.setNonBoolean\\(': {
```
Comment (lannka) [inline build-system/tasks/presubmit-checks.js]: I actually think all storage usage should be reviewed, including set boolean.

We might introduce a rule to check `storageForDoc`

File: src/service/storage-impl.js [hunk 2]
```diff
@@ -82,14 +82,25 @@ export class Storage {
...
+  }
+
+  /**
+   * Saves the value of the specified property. Returns the promise that's
+   * resolved when the operation completes.
+   * @param {string} name
+   * @param {*} value
+   * @return {!Promise}
+   */
+  setNonBoolean(name, value) {
```
Comment (lannka) [inline src/service/storage-impl.js]: are we separating it out only for the purpose of presubmit check?

File: build-system/tasks/presubmit-checks.js (hunk 1 above)
Comment (zhouyx) [reply build-system/tasks/presubmit-checks.js]: we have the check already.

File: src/service/storage-impl.js (hunk 2 above)
Comment (zhouyx) [reply src/service/storage-impl.js]: yes

Comment (lannka) [reply src/service/storage-impl.js]: If we already review the usage of storage service in general, I don't see too much value to add another presubmit check for non-boolean value.

Comment (zhouyx) [reply src/service/storage-impl.js]: I asked before why we have the boolean value check, and the answer was that we don't want to run privacy review for every value stored. Which means that the non boolean value requires more strict privacy check and I believe another presubmit check is necessary.

Comment (lannka) [reply src/service/storage-impl.js]: Got you. Could you indicate that in the JS doc that we need stricter privacy check for non-boolean storage.

Comment (zhouyx) [reply src/service/storage-impl.js]: done

**Pull request merged**